go_to_office/lib/screens/admin/offices_list.dart

```diff
@@ -11,17 +13,33 @@ final List<Office> officesList = [
 ];
 
 class OfficesListPage extends StatefulWidget {
+  OfficesListPage({Key key, this.title, this.repository, this.adminName}) : super(key: key);
+
+  final String title;
+  final Repository repository;
+  final String adminName;
+
   @override
-  State<StatefulWidget> createState() => _OfficesListPageState();
+  State<StatefulWidget> createState() => _OfficesListPageState(repository, adminName);
 }
 
 class _OfficesListPageState extends State<OfficesListPage> {
+  _OfficesListPageState(this.repository, this.adminName);
+
+  final Repository repository;
+  final String adminName;
+
   @override
   Widget build(BuildContext context) {
     // Object officateAssreddress = {};
     return Scaffold(
-        appBar: AppBar(
-          title: Text(Strings.offices),
+        appBar: AppBar(title: Text(widget.title)),
+        drawer: NavDrawer(userType: Strings.loginAdmin, title: adminName,
+          callback: (String pressed) => {
+            if (pressed == Strings.sign_out_button) {
```

Comment:
should be extracted to method
The NavDrawer does the same implementation for all screens and relies on data stored on the firebase.auth

Reply:
going to be changed